OOM: Java heap space

Only saw this once (so far), but it happened during server startup -- with -Xmx and -Xms set to 1.2G, so there is no way memory was actually exhausted.
```java
Caused by: java.lang.OutOfMemoryError: Java heap space
	at java.util.ArrayList.<init>(ArrayList.java:152) ~[?:1.8.0_111]
	at oshi.software.os.mac.MacOperatingSystem.getCommandLine(MacOperatingSystem.java:226) ~[oshi-core-3.4.3.jar:3.4.3]
	at oshi.software.os.mac.MacOperatingSystem.getProcess(MacOperatingSystem.java:191) ~[oshi-core-3.4.3.jar:3.4.3]
	at oshi.software.os.mac.MacOperatingSystem.getProcesses(MacOperatingSystem.java:94) ~[oshi-core-3.4.3.jar:3.4.3]
	at org.ziptie.zap.bootstrap.PerformanceSampler.run(PerformanceSampler.java:145) ~[com-dancer-zap-bootstrap.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_111]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) ~[?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) ~[?:1.8.0_111]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_111]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_111]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_111]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_111]
	at com.somepackage.GlobalTimerElf$LoggableScheduledFuture.invoke(GlobalTimerElf.java:176)
```
Judging from the error, I suspect some exceedingly large value was returned and oshi tried to allocate an enormous array.

Strangely, an immediate restart of the application did not encounter this error.  Nor had prior server starts, or have subsequent server starts.

Noteworthy is that the machine *was* just updated to High Sierra within the past 48 hours.

--- a/oshi-core/src/main/java/oshi/software/os/mac/MacOperatingSystem.java
+++ b/oshi-core/src/main/java/oshi/software/os/mac/MacOperatingSystem.java
@@ -88,10 +88,11 @@ public class MacOperatingSystem extends AbstractOperatingSystem {
     public OSProcess[] getProcesses(int limit, ProcessSort sort) {
         List<OSProcess> procs = new ArrayList<>();
         int[] pids = new int[this.maxProc];
-        int numberOfProcesses = SystemB.INSTANCE.proc_listpids(SystemB.PROC_ALL_PIDS, 0, pids, pids.length * SystemB.INT_SIZE)
-                / SystemB.INT_SIZE;
+        int numberOfProcesses = SystemB.INSTANCE.proc_listpids(SystemB.PROC_ALL_PIDS, 0, pids,
+                pids.length * SystemB.INT_SIZE) / SystemB.INT_SIZE;
         for (int i = 0; i < numberOfProcesses; i++) {
-            // Handle off-by-one bug in proc_listpids where the size returned is: SystemB.INT_SIZE * (pids + 1)
+            // Handle off-by-one bug in proc_listpids where the size returned
+            // is: SystemB.INT_SIZE * (pids + 1)
             if (pids[i] == 0) {
                 continue;
             }
@@ -222,12 +223,18 @@ public class MacOperatingSystem extends AbstractOperatingSystem {
         // Fetch arguments
         if (0 != SystemB.INSTANCE.sysctl(mib, mib.length, procargs, size, null, 0)) {
             LOG.error("Failed syctl call: kern.procargs2, Error code: {}", Native.getLastError());
+            return "";
         }
         // Procargs contains an int representing total # of args, followed by a
         // null-terminated execpath string and then the arguments, each
         // null-terminated (possible multiple consecutive nulls),
         // The execpath string is also the first arg.
         int nargs = procargs.getInt(0);
+        // Sanity check
+        if (nargs < 0 || nargs > 1024) {
+            LOG.error("Nonsensical number of process arguments for pid {}: {}", pid, nargs);
+            return "";
+        }
         List<String> args = new ArrayList<>(nargs);
         // Skip first int (containing value of nargs)
         long offset = SystemB.INT_SIZE;

